src/modules/news/news.js

```diff
@@ -1,126 +1,168 @@
+        this.start();
+    }
+
+    async start() {
+        try {
+            await this._fetchData();
+            this._createTitle();
+            this._createForm();
+        } catch (error) {
+            new Notification().showError('Fetch news data error', error);
+        }
+    }
+
+    async _fetchData() {
+        this._topNews = await this._api.getTopNews('top-headlines').then((response) => {
+            let main = this._createContainer();
+            let data = response.articles;
+            data.forEach((article) => {
+                console.log(article);
+                main.appendChild(this._createCard(article));
+            });
+        });
     }
 
-    start() {
-        this._createTitle();
-        this._createForm();
-        this._createContainer();
+    async _searchData(keyword) {
+        this._searchNews = await this._api.getSearchedNews(keyword).then((response) => {
+            let main = this._createContainer();
+            let data = response.articles;
+            data.forEach((article) => {
+                console.log(article);
+                main.appendChild(this._createCard(article));
+            });
+        });
     }
 
     _createTitle() {
-        const mainTitle = document.createElement('h1');
-        mainTitle.setAttribute('class', 'news__title');
-        this.viewElement.appendChild(mainTitle);
-        mainTitle.innerText = 'The News';
+        this._title = createElement('h1', { class: 'news__title' }, null, 'The News');
+        this.viewElement.appendChild(this._title);
     }
 
-    //helper function that set multiple attributes of an element
-    _setAttributes(element, attributes) {
-        for (let key in attributes) {
-            element.setAttribute(key, attributes[key]);
+    _searchKeyword() {
+        let newKeyword = document.querySelector('.news__input').value;
+        let container = document.querySelector('.news__container');
+        if (container && newKeyword) {
+            this.viewElement.removeChild(container);
+            this._searchData(newKeyword);
         }
     }
 
     _createForm() {
-        const form = document.createElement('form');
-        const label = document.createElement('label');
-        const inputText = document.createElement('input');
-        const inputSubmit = document.createElement('input');
-
-        form.setAttribute('class', 'news__form');
-        label.setAttribute('for', 'news__label');
-        this._setAttributes(inputText, {
+        this._form = createElement('form', { class: 'news__form' });
+        this._form.append(
+            this._createLabel(),
+            this._createInput(),
+            this._createButton('news__search--button', 'search', {
+                click: this._searchKeyword.bind(this),
+            })
+        );
+        this.viewElement.appendChild(this._form);
+    }
+
+    _createLabel() {
+        this._label = createElement('label', { for: 'news__label' });
+        return this._label;
```

Comment:
I think you can just return createElement('label', { for: 'news__label' });

We dont use this._label anywhere